public/login.php

```diff
@@ -50,28 +53,24 @@
         $csrfToken = bin2hex(random_bytes(16));
     }
 
-    session_name('CDash');
-    session_cache_limiter(@$SessionCachePolicy);
-    session_set_cookie_params($CDASH_COOKIE_EXPIRATION_TIME);
-    @ini_set('session.gc_maxlifetime', $CDASH_COOKIE_EXPIRATION_TIME + 600);
-    session_start();
-    $sessionArray = array('csrfToken' => $csrfToken);
+    // TODO: exists to satisfy a couple of tests and should be removed with extreme prejudice asap
+    if (!isset($SessionCachePolicy)) {
+        $SessionCachePolicy = Session::CACHE_PRIVATE_NO_EXPIRE;
+    }
 
-    $_SESSION['cdash'] = $sessionArray;
+    $session->start($SessionCachePolicy);
+    $session->setSessionVar('cdash', ['csrfToken' => $csrfToken]);
     LoginForm($loginerror); // display login form
     $session_OK = 0;
 } else {
     // authentication was successful
-    session_regenerate_id();
+    // $session->regenerateId();
```

Comment:
No need to regenerate the session ID here anymore?

Reply:
No, this should be here, good catch!